### PR TITLE
AuRa improvements

### PR DESCRIPTION
**Pull request by bkchr**
Hot and fresh AuRa improvements. This pr does the following:

- Move code belonging to the import queue etc to import_queue.rs
- Introduce `ImportQueueParams` and `StartAuraParams` structs to make
it more easier to understand what parameters we pass to AuRa.
- Introduce `CheckForEquivocation` to tell AuRa if it should check for
equivocation on block import. This is required for parachains, because
they are allowed to equivocate when they build two blocks for the same
slot, but for different relay chain parents.

### CODE REVIEW

File: client/consensus/aura/src/import_queue.rs [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
```
Comment (andresilva) [inline client/consensus/aura/src/import_queue.rs:18]:
```suggestion

//! Module implementing the logic for verifying and importing AuRa blocks.

```

**Review by andresilva — APPROVED** [review]

**Review by octol — APPROVED** [review]
Looks good!

File: bin/node-template/node/src/service.rs [hunk 2]
```diff
@@ -289,15 +295,18 @@ pub fn new_light(mut config: Configuration) -> Result<TaskManager, ServiceError>
...
-		sc_consensus_aura::slot_duration(&*client)?,
-		aura_block_import,
-		Some(Box::new(grandpa_block_import)),
-		client.clone(),
-		InherentDataProviders::new(),
-		&task_manager.spawn_essential_handle(),
-		config.prometheus_registry(),
-		sp_consensus::NeverCanAuthor,
+	let import_queue = sc_consensus_aura::import_queue::<AuraPair, _, _, _, _, _>(
+	ImportQueueParams {
```
Comment (octol) [inline bin/node-template/node/src/service.rs:299]: Indentation?

Comment (bkchr) [reply bin/node-template/node/src/service.rs]: Yeah :D TY

**Pull request merged**